### PR TITLE
model class tests

### PR DESCRIPTION
**Pull request by samuel-tate**

### CODE REVIEW

File: tests/compendia_tests.py
Comment (kurtwheeler) [inline tests/compendia_tests.py:109]: I think I like this testing strategy! However this looks fairly manual to write out for every field of every object.

I think this might be a bit easier:
```python
def assertCompendia(self, actual: Compendia, expected: dict):
  for key, value in expected.items():
    self.assertEqual(getattr(actual, key), value)
```

**Review by kurtwheeler — COMMENTED** [review]
This is looking good so far! I left one comment about the happy path test, but I think you're missing tests for the unhappy paths:
* What happens if the server is down and returns a 500?
* What happens if the user requests an id that doesn't exist?
* What happens if the user puts in too restrictive of filters and nothing is returned?
* What happens if the user provides the wrong data types to the filters or a filter that doesn't exist?

File: tests/compendia_tests.py
Comment (kurtwheeler) [inline tests/compendia_tests.py:143]: We should return a meaningful exception type with a descriptive/helpful message.

**Review by Philinphiladelphia — COMMENTED** [review]
Looking good to me. Nice testing strategy! Waiting on David's review for the approval.

File: pyrefinebio/__init__.py
Comment (kurtwheeler) [inline pyrefinebio/__init__.py:20]: Missing newline!

File: pyrefinebio/exceptions.py
Comment (kurtwheeler) [inline pyrefinebio/exceptions.py:25]: Missing newline!

File: pyrefinebio/__init__.py
Comment (kurtwheeler) [inline pyrefinebio/__init__.py:16]: Oh I just noticed this, but I prefer to stick to absolute rather than relative imports. The biggest reason is that if we ever wanted to move a file, we'd have to change all the imports as well with relative imports.

File: pyrefinebio/http.py
Comment (kurtwheeler) [inline pyrefinebio/http.py:5]: Same comment here about absolute rather than relative imports.

File: pyrefinebio/http.py
Comment (kurtwheeler) [inline pyrefinebio/http.py:40]:
```suggestion
            raise ServerError()
```

File: tests/custom_assertions.py
Comment (kurtwheeler) [inline tests/custom_assertions.py:15]:
```suggestion
            actual_value = getattr(actual, key)
            for key, expected_value in expected.items():
                if type(expected_value) is list:
                    if len(expected_value) != len(actual_value):
```

You call `getattr(actual, key)` quite a few times. It might be easier to read this function if you declare it as a variable and then reference that throughout the rest of it.

File: tests/custom_assertions.py
Comment (kurtwheeler) [inline tests/custom_assertions.py:44]: This function has some pretty deep nesting. You could remove at least one level of nesting by moving this check to the start of the function. If the assertion gets raised the rest of the function won't run so it doesn't even need to be in an `if`.

File: tests/mocks.py
Comment (kurtwheeler) [inline tests/mocks.py:14]: Missing newline!

File: tests/test_experiment.py
Comment (kurtwheeler) [inline tests/test_experiment.py:197]: You can take this TODO out!

**Review by kurtwheeler — CHANGES_REQUESTED** [review]
This looks great! So comprehensive! I love it!

Most of the comments I left are pretty small, except for the relative imports. If you want you can make that change in a subsequent PR.

File: tests/custom_assertions.py
Comment (kurtwheeler) [inline tests/custom_assertions.py:28]:
```suggestion
                if len(value) != len(actual_value):
```

**Review by kurtwheeler — APPROVED** [review]
I left one tiny suggestion but this looks great! Nice work!!!

**Pull request merged**
